sound/soc/sof/sof-priv.h

```diff
@@ -198,6 +198,8 @@ struct snd_sof_dsp_ops {
 	/* misc */
 	int (*get_bar_index)(struct snd_sof_dev *sdev,
 			     u32 type); /* optional */
+	int (*get_mailbox_offset)(struct snd_sof_dev *sdev);
```

Comment:
so this is mandatory then?

Reply:
it's actually mandatory only if the common loader code is used, so we should mention this.

Reply:
@plbossart ack.

I have marked Ranjani's comments as resolved perhaps thats the reason you cannot see them anymore. Mostly the comments were about style and moving the functions from utils.c to loader.c which I fixed in version 2 of the patch.

Reply:
@dbaluta can you add '/* mandatory */' comment as @plbossart mentioned please?

I can imagine that we should change sof_fw_ready() to be like this in the future:
```
/* check for ABI compatibility and create memory windows on first boot */
- int sof_fw_ready(struct snd_sof_dev *sdev, u32 msg_id)
+ int sof_fw_ready(struct snd_sof_dev *sdev, u32 offset)
{
	struct sof_ipc_fw_ready *fw_ready = &sdev->fw_ready;
-	int offset;
	int bar;
	int ret;

+      if (offset) {
-	/* mailbox must be on 4k boundary */
-	offset = snd_sof_dsp_get_mailbox_offset(sdev);
-	if (offset < 0) {
-		dev_err(sdev->dev, "error: have no mailbox offset\n");
-		return offset;
-	}
+              /* no valid offset from fw_ready message, use the default one */
+	        offset = snd_sof_dsp_get_mailbox_offset(sdev);
+	        if (offset < 0) {
+		        dev_err(sdev->dev, "error: have no mailbox offset\n");
+		        return offset;
+	        }
+        } else {
+                /* mailbox must be on 4k boundary */
+                offset <= 12;
+        }
```